internal/server/repository_worker.go

```diff
@@ -300,17 +298,16 @@ func (r *Repository) UpsertWorkerStatus(ctx context.Context, worker *Worker, opt
 			workerClone.PublicId = workerId
 			switch {
 			case opts.withKeyId != "":
-				// This case goes first in case we want to relax the restriction
-				// around both name and key ID being supplied to account for
-				// e.g. config processing bugs. In that case, a key ID being
-				// supplied should be a clear indicator that we are working with
-				// a PKI workerClone, and a lack of one a clear indication we are
+				// This case goes first because a key ID being supplied should
+				// be a clear indicator that we are working with a PKI
+				// workerClone, and a lack of one a clear indication we are
 				// working with a KMS workerClone.
 				//
 				// Note: unlike in the below case, this purposefully leaves out
-				// "description" since we want description changes for PKI-based
-				// workers to come via API only. We can't really guard on this
-				// in the DB so we need to be sure to not include it here.
+				// "description" since we want description changes for (non
+				// KMS-PKI) PKI-based workers to come via API only. We can't
+				// really guard on this in the DB so we need to be sure to not
+				// include it here.
 				n, err := w.Update(ctx, workerClone, []string{"address", "ReleaseVersion", "OperationalState"}, nil)
 				if err != nil {
 					return errors.Wrap(ctx, err, op, errors.WithMsg("unable to update status of pki worker"))
```

Comment:
Shouldn't this line be changed since the worker might be pki-kms?

Reply:
I don't think so -- but I may be wrong. The reason is that the actual registration for KMS-PKI workers happens in repository_workerauth in StoreNodeInformationTx and there we upsert the description that came from the worker.

The reason we may want to allow updates via this mechanism that you're asking about is specifically for KMS-PKI workers that change the description in their config file and SIGHUP. But I'm not sure we really want to optimize for that case as opposed to requiring a worker restart. So I think we want to keep it as is.

Make sense/agree, or disagree?